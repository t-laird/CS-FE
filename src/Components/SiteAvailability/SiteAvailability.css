.SiteAvailability {
  padding-top: 40px;
  background: white; }
  .SiteAvailability h3 {
    text-align: center; }
  .SiteAvailability .availabilityDetails {
    display: grid;
    justify-items: center;
    margin: 0px auto; }
    .SiteAvailability .availabilityDetails h3 {
      display: flex;
      justify-content: space-between;
      align-items: center;
      width: 600px; }
      .SiteAvailability .availabilityDetails h3 span {
        padding: 0px 15px;
        font-size: 30px;
        color: #02A364; }
    .SiteAvailability .availabilityDetails form {
      display: grid;
      height: 50px;
      grid-template-columns: 1fr 200px;
      margin: 0px auto; }
      .SiteAvailability .availabilityDetails form input {
        font-size: 20px; }
        .SiteAvailability .availabilityDetails form input:focus {
          outline: none; }
      .SiteAvailability .availabilityDetails form input[type="submit"] {
        background: white;
        border: 1px solid black;
        transition: 0.3s; }
        .SiteAvailability .availabilityDetails form input[type="submit"]:hover {
          color: white;
          background: #02A364; }
    .SiteAvailability .availabilityDetails span {
      text-align: center;
      padding-top: 15px;
      display: flex;
      align-items: center;
      justify-content: center;
      font-size: 24px; }
      .SiteAvailability .availabilityDetails span span {
        color: #02A364;
        padding: 0px 0px 0px 15px;
        font-size: 200%; }
