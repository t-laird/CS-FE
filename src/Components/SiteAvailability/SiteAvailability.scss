.SiteAvailability {
  padding-top: 40px;
  background: white;

  h3 {
    text-align: center;
  }

  .availabilityDetails {
    display: grid;
    justify-items: center;
    margin: 0px auto;
    h3 {
      // text-align: center;
      display: flex;
      justify-content: space-between;
      align-items: center;
      width: 600px;
      span {
        padding: 0px 15px;
        font-size: 30px;
        color: #02A364;
      }
    }
    form {
      display: grid;
      height: 50px;
      grid-template-columns: 1fr 200px;
      margin: 0px auto;
  
      input {
        font-size: 20px;
        &:focus {
          outline: none;
        }
      }
  
      input[type="submit"] {
        background: white;
        border: 1px solid black;
        transition: 0.3s;
        &:hover {
          color: white;
          background: #02A364;
        }
      }
    }

    span {
      text-align: center;
      padding-top: 15px;
      display: flex;
      align-items: center;
      justify-content: center;
      font-size: 24px;
      span {
        color: #02A364;
        padding: 0px 0px 0px 15px;
        font-size: 200%;
      }
    }
  }
}