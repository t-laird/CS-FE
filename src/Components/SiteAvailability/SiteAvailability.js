import React, { Component } from 'react';
import Availability from '../Availability/Availability';
import './SiteAvailability.css';

class SiteAvailability extends Component {
  constructor() {
    super();
    this.state = {
      gap: 'unset'
    };
  }

  handleUpdate = (e) => {
    e.preventDefault();
    this.setState({ gap: parseInt(this.gapSize.value, 10) });
    this.gapSize.focus();
  }

  validReservationData = () => {  
    const { reservations, search, campsites } = this.props;
    const gapInfo = (
      isNaN(this.state.gap) || 
      this.state.gap > 6 || 
      this.state.gap < 0 
    ) ? <span>Input a gap rule size between 0-5 to highlight reservation availability.</span>
      : <span>Gap Size is <span>{this.state.gap}</span></span>
    
    if (!(Object.keys(search).length)) {
      return (
        <h3>Input Reservation Dates to See Availability</h3>
      );
    } else {
      return (
        <div className="availabilityDetails">
          <h3>Availability from <span>{search.startDate}</span> to <span>{search.endDate}</span></h3>
          <form onSubmit={this.handleUpdate}>
            <input type="text" ref={(input) => this.gapSize=input } placeholder="Enter the Desired Gap Size"/>
            <input type="submit" value="Highlight Gap Size" />
          </form>
          {gapInfo}
          <Availability gap={this.state.gap} campsites={campsites}/>
        </div>
      )
    }
  }


  render() {
    return (
      <div className="SiteAvailability">
        {this.validReservationData()}
      </div>
    )
  }
}
export default SiteAvailability;
