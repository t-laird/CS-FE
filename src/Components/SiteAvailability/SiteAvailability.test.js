import React from 'react';
import { shallow } from 'enzyme';
import SiteAvailability from './SiteAvailability';

describe('SiteAvailability component tests', () => {
  let renderedSite;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      "campsites": { "1": { "name": "Grizzly Adams Adventure Cabin", "gaps": { "0": true, "1": true, "2": true, "3": true, "4": true, "5": true } } },
      "reservations": [ { "campsiteId": 1, "startDate": "2016-06-01", "endDate": "2016-06-03" }],
      "search": { "startDate": "2016-06-04", "endDate": "2016-06-06" },
    }
    renderedSite = shallow(<SiteAvailability {...mockProps} />);
  });

  it('Should match the snapshot', () => {
    expect(renderedSite).toMatchSnapshot();
  });

  it('Should have the expected default elements', () => {
    expect(renderedSite.find('.SiteAvailability').length).toEqual(1);
  });

  it('Should render the availability details if there are reservations', () => {
    expect(renderedSite.find('.availabilityDetails').length).toEqual(1);
    expect(renderedSite.find('form').length).toEqual(1);
    expect(renderedSite.find('Availability').length).toEqual(1);
  });

  it('Should not render availability details if information has not yet been passed', () => {
    const emptyMockProps = { reservations: [], search: {}, campsites: {}};
    const emptyDataWrapper = shallow(<SiteAvailability {...emptyMockProps} />);
  });
});