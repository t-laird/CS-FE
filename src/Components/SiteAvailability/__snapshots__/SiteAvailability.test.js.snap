// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteAvailability component tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteAvailability
    campsites={
        Object {
            "1": Object {
              "gaps": Object {
                "0": true,
                "1": true,
                "2": true,
                "3": true,
                "4": true,
                "5": true,
              },
              "name": "Grizzly Adams Adventure Cabin",
            },
          }
    }
    reservations={
        Array [
            Object {
              "campsiteId": 1,
              "endDate": "2016-06-03",
              "startDate": "2016-06-01",
            },
          ]
    }
    search={
        Object {
            "endDate": "2016-06-06",
            "startDate": "2016-06-04",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="availabilityDetails"
>
        <h3>
                Availability from 
                <span>
                        2016-06-04
                </span>
                 to 
                <span>
                        2016-06-06
                </span>
        </h3>
        <form
                onSubmit={[Function]}
        >
                <input
                        placeholder="Enter the Desired Gap Size"
                        type="text"
                />
                <input
                        type="submit"
                        value="Highlight Gap Size"
                />
        </form>
        <span>
                Input a valid gap size between 0-5 to highlight reservation availability.
        </span>
        <Availability
                campsites={
                        Object {
                                "1": Object {
                                  "gaps": Object {
                                    "0": true,
                                    "1": true,
                                    "2": true,
                                    "3": true,
                                    "4": true,
                                    "5": true,
                                  },
                                  "name": "Grizzly Adams Adventure Cabin",
                                },
                              }
                }
                gap="unset"
        />
</div>,
      "className": "SiteAvailability",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Availability from 
            <span>
                        2016-06-04
            </span>
             to 
            <span>
                        2016-06-06
            </span>
</h3>,
          <form
            onSubmit={[Function]}
>
            <input
                        placeholder="Enter the Desired Gap Size"
                        type="text"
            />
            <input
                        type="submit"
                        value="Highlight Gap Size"
            />
</form>,
          <span>
            Input a valid gap size between 0-5 to highlight reservation availability.
</span>,
          <Availability
            campsites={
                        Object {
                                    "1": Object {
                                      "gaps": Object {
                                        "0": true,
                                        "1": true,
                                        "2": true,
                                        "3": true,
                                        "4": true,
                                        "5": true,
                                      },
                                      "name": "Grizzly Adams Adventure Cabin",
                                    },
                                  }
            }
            gap="unset"
/>,
        ],
        "className": "availabilityDetails",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Availability from ",
              <span>
                2016-06-04
</span>,
              " to ",
              <span>
                2016-06-06
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Availability from ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2016-06-04",
              },
              "ref": null,
              "rendered": "2016-06-04",
              "type": "span",
            },
            " to ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2016-06-06",
              },
              "ref": null,
              "rendered": "2016-06-06",
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                placeholder="Enter the Desired Gap Size"
                type="text"
/>,
              <input
                type="submit"
                value="Highlight Gap Size"
/>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Enter the Desired Gap Size",
                "type": "text",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Highlight Gap Size",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Input a valid gap size between 0-5 to highlight reservation availability.",
          },
          "ref": null,
          "rendered": "Input a valid gap size between 0-5 to highlight reservation availability.",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "campsites": Object {
              "1": Object {
                "gaps": Object {
                  "0": true,
                  "1": true,
                  "2": true,
                  "3": true,
                  "4": true,
                  "5": true,
                },
                "name": "Grizzly Adams Adventure Cabin",
              },
            },
            "gap": "unset",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="availabilityDetails"
>
          <h3>
                    Availability from 
                    <span>
                              2016-06-04
                    </span>
                     to 
                    <span>
                              2016-06-06
                    </span>
          </h3>
          <form
                    onSubmit={[Function]}
          >
                    <input
                              placeholder="Enter the Desired Gap Size"
                              type="text"
                    />
                    <input
                              type="submit"
                              value="Highlight Gap Size"
                    />
          </form>
          <span>
                    Input a valid gap size between 0-5 to highlight reservation availability.
          </span>
          <Availability
                    campsites={
                              Object {
                                        "1": Object {
                                          "gaps": Object {
                                            "0": true,
                                            "1": true,
                                            "2": true,
                                            "3": true,
                                            "4": true,
                                            "5": true,
                                          },
                                          "name": "Grizzly Adams Adventure Cabin",
                                        },
                                      }
                    }
                    gap="unset"
          />
</div>,
        "className": "SiteAvailability",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Availability from 
              <span>
                            2016-06-04
              </span>
               to 
              <span>
                            2016-06-06
              </span>
</h3>,
            <form
              onSubmit={[Function]}
>
              <input
                            placeholder="Enter the Desired Gap Size"
                            type="text"
              />
              <input
                            type="submit"
                            value="Highlight Gap Size"
              />
</form>,
            <span>
              Input a valid gap size between 0-5 to highlight reservation availability.
</span>,
            <Availability
              campsites={
                            Object {
                                          "1": Object {
                                            "gaps": Object {
                                              "0": true,
                                              "1": true,
                                              "2": true,
                                              "3": true,
                                              "4": true,
                                              "5": true,
                                            },
                                            "name": "Grizzly Adams Adventure Cabin",
                                          },
                                        }
              }
              gap="unset"
/>,
          ],
          "className": "availabilityDetails",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Availability from ",
                <span>
                  2016-06-04
</span>,
                " to ",
                <span>
                  2016-06-06
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Availability from ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2016-06-04",
                },
                "ref": null,
                "rendered": "2016-06-04",
                "type": "span",
              },
              " to ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2016-06-06",
                },
                "ref": null,
                "rendered": "2016-06-06",
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  placeholder="Enter the Desired Gap Size"
                  type="text"
/>,
                <input
                  type="submit"
                  value="Highlight Gap Size"
/>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Enter the Desired Gap Size",
                  "type": "text",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Highlight Gap Size",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Input a valid gap size between 0-5 to highlight reservation availability.",
            },
            "ref": null,
            "rendered": "Input a valid gap size between 0-5 to highlight reservation availability.",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "campsites": Object {
                "1": Object {
                  "gaps": Object {
                    "0": true,
                    "1": true,
                    "2": true,
                    "3": true,
                    "4": true,
                    "5": true,
                  },
                  "name": "Grizzly Adams Adventure Cabin",
                },
              },
              "gap": "unset",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
